test/unittest/tests/block/test_block_api_misc.cpp

```diff
@@ -0,0 +1,257 @@
+     *   //Test that the block manager's addr_invalid method returns an error when checking if the
+     *   //address cookie is valid.
+     *   REQUIRE(test_addr_invalid(s, &bm, 512, 1024, 12345) == WT_PANIC);
+     *
+     * }
+     */
+
+    // Cleanup for block created during block manager initialization.
+    REQUIRE(__wt_block_close(s, bm.block) == 0);
+    REQUIRE(__wt_block_manager_drop(s, file_path.c_str(), false) == 0);
+}
+
+TEST_CASE("Block manager addr string", "[block_api_misc]")
+{
+    // Build Mock session, this will automatically create a mock connection.
+    std::shared_ptr<mock_session> session = mock_session::build_test_mock_session();
+
+    WT_BM bm;
+    WT_CLEAR(bm);
+    auto path = std::filesystem::current_path();
+    std::string file_path(path.string() + "/test.wt");
+    setup_bm(session, &bm, file_path);
+    WT_SESSION_IMPL *s = session->get_wt_session_impl();
+
+    SECTION("Test addr string with non-zero values")
+    {
+        // (512, 1024, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 1024, 12345, "[0: 512-1536, 1024, 12345]");
+    }
+
+    SECTION("Test addr string with zero values")
+    {
+        // (0, 0, 0) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 0, 0, 0, "[0: 0-0, 0, 0]");
+    }
+
+    SECTION("Test addr string with zero size")
+    {
+        // (512, 0, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 0, 12345, "[0: 0-0, 0, 0]");
+    }
+
+    REQUIRE(__wt_block_close(s, bm.block) == 0);
+}
+
+TEST_CASE("Block header", "[block_api_misc]")
+{
+    // Declare a block manager and set it up so that we can use its legal API methods.
+    WT_BM bm;
+    WT_CLEAR(bm);
+    __wti_bm_method_set(&bm, false);
+
+    REQUIRE(bm.block_header(&bm) == (u_int)WT_BLOCK_HEADER_SIZE);
+}
+
+TEST_CASE("Block manager is mapped", "[block_api_misc]")
+{
+    // Declare a block manager and set it up so that we can use its legal API methods.
+    WT_BM bm;
+    WT_CLEAR(bm);
+    __wti_bm_method_set(&bm, false);
+
+    SECTION("Test block manager is mapped")
+    {
+        uint8_t i;
+        bm.map = &i;
+        REQUIRE(bm.is_mapped(&bm, nullptr) == true);
+    }
+
+    SECTION("Test block manager is not mapped")
+    {
+        bm.map = nullptr;
+        REQUIRE(bm.is_mapped(&bm, nullptr) == false);
+    }
+}
+
+TEST_CASE("Block manager size and stat", "[block_api_misc]")
+{
+    // Build Mock session, this will automatically create a mock connection.
+    std::shared_ptr<mock_session> session = mock_session::build_test_mock_session();
+
+    WT_BM bm;
+    WT_CLEAR(bm);
+    auto path = std::filesystem::current_path();
+    std::string file_path(path.string() + "/test.wt");
+    setup_bm(session, &bm, file_path);
+    WT_SESSION_IMPL *s = session->get_wt_session_impl();
+
+    wt_off_t prev_size;
+    REQUIRE(bm.size(&bm, s, &prev_size) == 0);
+
+    // Test that the bm->stat method updates statistics correctly after initializing the block
```

Comment:
Lines 235-237 could be SECTION("Test that the bm->stat method updates statistics correctly after initializing the block manager");

Lines 232-233 moved after 237 and lines 239-253 could be SECTION("Test that the bm->data method updated statistics correctly after doing a write and that the block size increased.");